### DISCUSSION TITLE
Possible to create an entire stack using the API?

### DISCUSSION THREAD

**Discussion by mike-hogan** [Budibase/budibase#12521] [General]
I am consider writing a small saas offering that solves for businesses that are based on bookings and appointments.  I know there are tons of these already, but there are features missing that my (small) customer base is looking for.

I am considering doing this as a companion app to Budibase.  Budibase solves so much of the base layer of features, that I can focus on the differentiated part my customers want - a slick booking experience for *their* customers

I know Budibase has templates, but I think I can provide a better experience for my customers if I can build the entire application using the API.

Some questions:

1) Does the API support this level of ground-up app construction?
2) How might I handle migrations?  Lets say I want to migrate my customer so that phone numbers can be captured -> I'd like that applied to all apps
3) is this approach to companion saas commercially acceptable to the Budibase team?

Thx!
Mike

**Comment by ConorWebb96** [comment]
Hey @mike-hogan,

1. Anything you can do within Budibase e.g. form submissions updates etc can be done through the [public API](https://docs.budibase.com/reference/appcreate).
2. Migrations might be a bit tricky, as you will likely have to duplicate the custom users table across your apps. This means the more apps you add the harder this would likely be to manage in my opinion. As the in-built users table doesn't support additional fields outside of first, last name and email address.
3. I don't believe there would be an issue with this, you can use Budibase to do whatever you wish, its completely up to the user.

Hopefully this answers your questions. 🚀
